### PR TITLE
Initial setup for docs/ directory

### PR DESCRIPTION
**Pull request by lamberta**
This moves some files around for the docs/ directory to work with the site. This merges into the master branch.

* Sphinx files are moved into `dev_tools/docs/sphinx/`, though they should be removed at some point.
* Notebooks moved from `examples/` into `docs/tutorials` to be consistent with the other projects. examples/ directory should probably be removed at some point, or used for more "real world" demos. I'll send a follow-up PR to format notebooks and remove outputs so they're tested with the docs infra.
* Run `format-incremental --apply` on sphinx/conf.py to pass CI.

Context: b/159747117

### CODE REVIEW

**Comment by lamberta** [comment]
When running `./check/pytest` locally I only see failures in `src/openfermion/chem/pubchem_test.py`:

```
== short test summary info ==
FAILED src/openfermion/chem/pubchem_test.py::OpenFermionPubChemTest::test_helium - pubchempy.ServerError: 'PUGREST.ServerError'
FAILED src/openfermion/chem/pubchem_test.py::OpenFermionPubChemTest::test_none - pubchempy.ServerError: 'PUGREST.ServerError'
FAILED src/openfermion/chem/pubchem_test.py::OpenFermionPubChemTest::test_water - pubchempy.ServerError: 'PUGREST.ServerError'
FAILED src/openfermion/chem/pubchem_test.py::OpenFermionPubChemTest::test_water_2d - pubchempy.ServerError: 'PUGREST.ServerError'
== 4 failed, 2066 passed, 131 warnings in 226.45s (0:03:46)  ==
```

**Comment by ncrubin** [comment]
@lamberta Thanks for doing this.  I reinitialized the test and the issue with the NIH servers seems resolved.  This is another good reminder that we had intended to change that functionality.

File: docs/install.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+python -m pip install -e .
+```
+
+## Library install
+
+To install the latest PyPI release as a library (in user mode):
+
+```bash
+python -m pip install --user openfermion
+```
```
Comment (ncrubin) [inline docs/install.md:41]: Could we add a build the docs section

# Build the docs

>>> line to build the docs

Then I think we're good to go!

Comment (lamberta) [reply docs/install.md:78]: Done, PTAL. Added a small section about previewing narrative docs in GitHub/Colab, and included a section for building Markdown for API reference pages.

**Comment by ncrubin** [comment]
Thank you @lamberta !

**Pull request merged**